File: backend/app/schemas/asset.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.schemas.service import ServiceRead


class AssetRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    project_id: uuid.UUID
    target_id: uuid.UUID
    scan_id: uuid.UUID
    host: str
    created_at: datetime
    services: list[ServiceRead] = []

File: frontend/src/lib/types/asset.ts
/**
 * Types mirroring the backend's Asset Pydantic schema
 * (see backend Phase 3 Task 5 `AssetRead`).
 */

import type { ServiceRead } from "./service";

export interface AssetRead {
  id: string;
  project_id: string;
  target_id: string;
  scan_id: string;
  host: string;
  created_at: string;
  services: ServiceRead[];
}
